src/fireedge/src/modules/components/Forms/ServiceTemplate/CreateForm/Steps/Roles/dropdowns/network.js

```diff
@@ -0,0 +1,254 @@
+    }
+
+    update(nicIndex, updatedNIC)
+  }
+
+  const handleFloatingIp = (rowId, toggle, type) => {
+    const nicIndex = NICs?.findIndex((nic) => nic?.NETWORK_ID === rowId)
+
+    if (nicIndex === -1) return
+
+    const updatedNIC = { ...NICs[nicIndex] }
+
+    if (!toggle) {
+      delete updatedNIC[type]
+    } else {
+      updatedNIC[type] = 'yes'
+    }
+
+    update(nicIndex, updatedNIC)
+  }
+
+  // Clears floating IP fields
+  useEffect(() => {
+    if (!isVr) {
+      replace(
+        NICs?.map(({ FLOATING_IP, FLOATING_ONLY, ...nic }) => ({ ...nic }))
+      )
+    }
+  }, [isVr])
+
+  const columns = [
+    {
+      field: 'name',
+      headerName: 'Network',
+      flex: isVr ? 8 / 24 : 12 / 24,
+    },
+
+    ...(isVr
+      ? [
+          {
+            field: 'floating_ip',
+            headerName: 'Floating IP',
+            flex: 4 / 24,
+            renderCell: (params) => {
+              if (!isVr) return null
+
+              return (
+                <Checkbox
+                  onChange={(_, value) =>
+                    handleFloatingIp(params.row.id, value, 'FLOATING_IP')
+                  }
+                  checked={
+                    NICs?.find((nic) => nic?.NETWORK_ID === params?.row?.id)
+                      ?.FLOATING_IP === 'yes'
+                  }
+                />
+              )
+            },
+          },
+          {
+            field: 'floating_only',
+            headerName: 'Floating Only',
+            flex: 4 / 24,
+            renderCell: (params) => {
+              if (!isVr) return null
+
+              return (
+                <Checkbox
+                  onChange={(_, value) =>
+                    handleFloatingIp(params.row.id, value, 'FLOATING_ONLY')
+                  }
+                  checked={
+                    NICs?.find((nic) => nic?.NETWORK_ID === params?.row?.id)
+                      ?.FLOATING_ONLY === 'yes'
+                  }
+                />
+              )
+            },
+          },
+        ]
+      : []),
+
+    {
+      field: 'NIC_ALIAS',
+      headerName: 'As NIC Alias',
+      flex: isVr ? 8 / 24 : 12 / 24,
+      renderCell: (params) => {
+        const isSelected = NICs?.find(
+          (NIC) => NIC?.NETWORK_ID === params?.row.id
+        )
+
+        const availableAliases = NICs.filter(
+          (NIC) =>
+            NIC.NETWORK_ID !== params.row.id &&
+            !NICs?.some((nic) => nic?.NIC_ALIAS?.NETWORK_ID === params.row.id)
+        )
+
+        if (!isSelected || !availableAliases?.length) return null
+
+        return (
+          <Autocomplete
+            options={availableAliases}
+            getOptionLabel={(option) => option?.NETWORK_ID?.replace('$', '')}
+            value={
+              NICs.find((NIC) => NIC.NETWORK_ID === params.row.id)?.NIC_ALIAS ||
+              null
+            }
+            onChange={(_, newValue) => handleAlias(params.row.id, newValue)}
+            renderInput={(args) => (
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

JSX Component label not internationalized.